src/HDF5Msg.chpl

```diff
@@ -896,6 +885,186 @@ module HDF5Msg {
         }
     }
 
+    proc segarray_tohdfMsg(msgArgs: MessageArgs, st: borrowed SymTab) throws {
+        use C_HDF5.HDF5_WAR;
+        var mode: int = msgArgs.get("write_mode").getIntValue();
+
+        var filename: string = msgArgs.getValueOf("filename");
+        var file_format = msgArgs.get("file_format").getIntValue();
+        var group = msgArgs.getValueOf("dset");
+        var dType = str2dtype(msgArgs.getValueOf("dtype"));
+
+        var vals = st.lookup(msgArgs.getValueOf("values"));
+
+        var segs = st.lookup(msgArgs.getValueOf("segments"));
+        var segEntry:SymEntry = toSymEntry(toGenSymEntry(segs), int);
+
+        const objType = msgArgs.getValueOf("objType");
+
+        select file_format {
+            when SINGLE_FILE {
+                var f = prepFiles(filename, mode);
+                var file_id = C_HDF5.H5Fopen(f.c_str(), C_HDF5.H5F_ACC_RDWR, C_HDF5.H5P_DEFAULT);
+                if file_id < 0 { // HF5open returns negative value on failure
+                    C_HDF5.H5Fclose(file_id);
+                    var errorMsg = "Failure accessing file %s.".format(f);
+                    throw getErrorWithContext(
+                           msg=errorMsg,
+                           lineNumber=getLineNumber(),
+                           routineName=getRoutineName(), 
+                           moduleName=getModuleName(),
+                           errorClass="FileNotFoundError");
+                }
+
+                // create the group
+                validateGroup(file_id, f, group);
+
+                inline proc writeLocalSegArray(file_id: C_HDF5.hid_t, group: string, vals, valDim: int, segs, segDim: int, type t) throws {
```

Comment:
I would suggest pulling this function definition out to the module level instead of nesting to match the structure we use elsewhere.

Reply:
Function moved to module level